fe1-web/src/core/network/jsonrpc/__tests__/ExtendedJsonRpcRequest.test.ts

```diff
@@ -0,0 +1,29 @@
+import { describe, it } from '@jest/globals';
+
+import { ExtendedJsonRpcRequest } from '../ExtendedJsonRpcRequest';
+import { JsonRpcMethod } from '../JsonRpcMethods';
+import { JsonRpcRequest } from '../JsonRpcRequest';
+
+const mockAddress = 'some address';
+const mockChannel = 'some channel';
+const mockJsonRequest: Partial<JsonRpcRequest> = {
+  jsonrpc: 'some data',
+  method: JsonRpcMethod.BROADCAST,
+  params: { channel: mockChannel },
+};
+
+describe('ExtendedJsonRpcRequest', () => {
+  it('can create a new instance', () => {
+    const extendedRequest = new ExtendedJsonRpcRequest(
+      { receivedFrom: mockAddress },
+      mockJsonRequest,
+    );
+    expect(extendedRequest.request).toEqual(new JsonRpcRequest(mockJsonRequest));
+    expect(extendedRequest.receivedFrom).toEqual(mockAddress);
+  });
+
+  it('cannot create a new instance with missing receivedFrom parameter', () => {
+    const fn = () => new ExtendedJsonRpcRequest({}, mockJsonRequest);
+    expect(fn).toThrow();
```

Comment:
```suggestion
    expect(fn).toThrow(ProtocolError);
```
It might be better to ensure it is the right error.

Reply:
👍